crypto/txscript/src/data_stack.rs

```diff
@@ -218,7 +195,16 @@ impl<const LEN: usize> OpcodeData<SizedEncodeInt<LEN>> for Vec<u8> {
 
     #[inline]
     fn serialize(from: &SizedEncodeInt<LEN>) -> Result<Self, SerializationError> {
-        Ok(serialize_i64(&from.0))
+        // This is an optimzation for LEN 8. Without it, the function will fail when checking the serialized data length.
+        if LEN == 8 && from.0 == i64::MIN {
```

Comment:
I argue that marginal optimizations of an error execution path which will never actually happen, are meaningless. I'd suggest explaining in a comment over the condition below `if bytes.len() > LEN` that even for `LEN == 8`, this will fail for `u64::MIN` and that it is by design